folly/lang/ToAscii.h

```diff
@@ -56,17 +56,23 @@ using to_ascii_alphabet_upper = to_ascii_alphabet<true>;
 
 namespace detail {
 
+#if defined(_M_IX86)
```

Comment:
This is way more complicated than I would like.

IIUC, only MSVC has the property where the 64-bit intrinsic `__lzcnt64` is unavailable under 32-bit builds. For GNU-compatible compilers, the 64-bit builtin `__builtin_clzll` is implemented natively under 64-bit builds and emulated under 32-bit builds.